lib/Parse/Lexer.cpp

```diff
@@ -1477,6 +1477,29 @@ void Lexer::lexImpl() {
       } else {
         diagnose(CurPtr-1, diag::lex_invalid_character)
           .fixItReplaceChars(getSourceLoc(CurPtr-1), getSourceLoc(tmp), " ");
+
+        StringRef confusedChar;
+        if (codepoint >= 0x80) {
+          llvm::SmallString<64> buffer;
+          EncodeToUTF8(codepoint, buffer);
+          confusedChar = buffer.str();
+        } else {
+          confusedChar = StringRef(std::string(1, (char)codepoint));
```

Comment:
`std::string` constructs a temporary string.  `StringRef` creates a reference to that temporary object.  The object goes away at the end of this statement, and `confusedChar` refers to freed memory when it is used below.